### ISSUE TITLE
Unable to resolve DNS queries in bridge network mode on DSM

### ISSUE THREAD

**Issue by nicoelayda** [MatthewVance/unbound-docker#69]
When I create the standard container using the default configuration and Docker's default bridge network, I'm getting `SERVFAIL` trying to resolve DNS queries.

However, when I create the container and use the host network, it seems to work just fine. Is using the host network a requirement for this to work?

**Bridge mode setup:**

```sh
docker run --name my-unbound -d -p 53:53/udp -p 53:53/tcp --restart=always mvance/unbound:1.13.1
```

**`dig` output (bridge mode)**

```sh
❯ dig sigok.verteiltesysteme.net @192.168.1.220 -p 53

; <<>> DiG 9.10.6 <<>> sigok.verteiltesysteme.net @192.168.1.220
;; global options: +cmd
;; Got answer:
;; ->>HEADER<<- opcode: QUERY, status: SERVFAIL, id: 59180
;; flags: qr rd ra; QUERY: 1, ANSWER: 0, AUTHORITY: 0, ADDITIONAL: 1

;; OPT PSEUDOSECTION:
; EDNS: version: 0, flags:; udp: 1232
;; QUESTION SECTION:
;sigok.verteiltesysteme.net.	IN	A

;; Query time: 5 msec
;; SERVER: 192.168.1.220#53(192.168.1.220)
;; WHEN: Tue Jul 06 04:05:40 +08 2021
;; MSG SIZE  rcvd: 55
```

**Host mode setup:**

```sh
docker run --name my-unbound -d --net=host --restart=always mvance/unbound:1.13.1
```

**`dig` output (host mode)**

```sh
❯ dig sigok.verteiltesysteme.net @192.168.1.220 -p 53

; <<>> DiG 9.10.6 <<>> sigok.verteiltesysteme.net @192.168.1.220
;; global options: +cmd
;; Got answer:
;; ->>HEADER<<- opcode: QUERY, status: NOERROR, id: 64535
;; flags: qr rd ra ad; QUERY: 1, ANSWER: 1, AUTHORITY: 0, ADDITIONAL: 1

;; OPT PSEUDOSECTION:
; EDNS: version: 0, flags:; udp: 1232
;; QUESTION SECTION:
;sigok.verteiltesysteme.net.	IN	A

;; ANSWER SECTION:
sigok.verteiltesysteme.net. 300	IN	A	134.91.78.139

;; Query time: 645 msec
;; SERVER: 192.168.1.220#53(192.168.1.220)
;; WHEN: Tue Jul 06 04:00:10 +08 2021
;; MSG SIZE  rcvd: 71
```

Here's my bridge network config (from `docker network inspect bridge`):

```json
[
    {
        "Name": "bridge",
        "Id": "24cce2c17f87c48f0e1b94faa7c65a3612e2f54fdc00319512b629419c4d7e7d",
        "Created": "2021-07-04T07:06:55.445310109+08:00",
        "Scope": "local",
        "Driver": "bridge",
        "EnableIPv6": false,
        "IPAM": {
            "Driver": "default",
            "Options": null,
            "Config": [
                {
                    "Subnet": "172.17.0.0/16",
                    "Gateway": "172.17.0.1"
                }
            ]
        },
        "Internal": false,
        "Attachable": false,
        "Ingress": false,
        "ConfigFrom": {
            "Network": ""
        },
        "ConfigOnly": false,
        "Containers": {
            "28f7a1b6ec7e1652f5ae1caf3aee30381555b50ed4d1100b336321dfecb1fd1e": {
                "Name": "my-unbound",
                "EndpointID": "1ce9de0a2cbf044425d11511df435bf58ab05fe676947424fe7ec1922aa78a89",
                "MacAddress": "02:42:ac:11:00:03",
                "IPv4Address": "172.17.0.3/16",
                "IPv6Address": ""
            },
            "c4c40664a14915a3866b90f05224b1db2d475d55fa3f9c81ad59e8b7b3c3c7dc": {
                "Name": "portainer-ce",
                "EndpointID": "7f3fcfc269352339bc0730fce83bcc3a9943acda528827f1b7030a500116ce0b",
                "MacAddress": "02:42:ac:11:00:02",
                "IPv4Address": "172.17.0.2/16",
                "IPv6Address": ""
            }
        },
        "Options": {
            "com.docker.network.bridge.default_bridge": "true",
            "com.docker.network.bridge.enable_icc": "true",
            "com.docker.network.bridge.enable_ip_masquerade": "true",
            "com.docker.network.bridge.host_binding_ipv4": "0.0.0.0",
            "com.docker.network.bridge.name": "docker0",
            "com.docker.network.driver.mtu": "1500"
        },
        "Labels": {}
    }
]
```

**Comment by MatthewVance** [comment]
No, host network is not a requirement. Some Linux distros come with a DNS
service already running on 53. That can cause problem if you don't stop
that service first. However, I would expect a port conflict error if this
was the case.

Is 192.168.1.220 the IP of your host? The server with Docker installed?

**Comment by nicoelayda** [comment]
Yes, 192.168.1.220 is my host IP with Docker. I've confirmed that port 53 is free, there are no containers running other than Portainer and unbound.

I'm running this on a Synology NAS with DSM 7.0. I'll try tonight with a Raspberry Pi and see if I observe the same issue.

**Comment by nicoelayda** [comment]
Update: I've tried this on Ubuntu 20.04.2 LTS arm64 running on my RPi4 using pedantic87's container and it works perfectly using bridge networking.

Looks like this problem is specific to DSM.

**Comment by MatthewVance** [comment]
Thanks for checking and confirming.

**Comment by MatthewVance** [comment]
Closing this issue because it appears to be a local DSM configuration issue rather than a container issue. Others have it running on Synology without issue. For instance:
https://github.com/MatthewVance/unbound-docker/issues/35#issuecomment-607822002
https://github.com/MatthewVance/unbound-docker/issues/39#issue-613536209
https://github.com/MatthewVance/unbound-docker/issues/44#issuecomment-654978355

**Comment by nicoelayda** [comment]
Thanks for the links. I've narrowed down the problem to DNS-over-TLS not working on DSM for some reason.

If I change the default `forward-records.conf` to this, it works on bridge networking mode:

```
forward-zone:
    name: "."
    forward-tls-upstream: no
    forward-addr: 1.1.1.1@53
    forward-addr: 1.0.0.1@53
```

I didn't check further what the problem with DoT was, I just decided to deploy the container in a macvlan network with a dedicated IP, similar to the setup in your first link. When connected to a macvlan network, the container works as expected on DSM.
